openadapt/record.py

```diff
@@ -618,12 +631,29 @@ def record(
 
     # TODO: discard events until everything is ready
 
+    collect_stats()
+
     try:
         while True:
             time.sleep(1)
     except KeyboardInterrupt:
         terminate_event.set()
 
+    collect_stats()
+
+    stats = performance_snapshots[-1].compare_to(performance_snapshots[-2], 'lineno')
+
+    for stat in stats[:3]:
```

Comment:
Why do we start at stats[:3]?

Reply:
just `stats` would show every single memory allocation and there are a lot, so it fills up the logging with a lot of insignificant memory allocations that only take a few bytes. I've defined a constant `NUM_MEMORY_STATS` for the 3, so we could change that to 5 or 10 or however many stats we want to see.